
#BIOL7263

This is the home for all of my work for my favorite class!

# Here are my notes from:

 [Day 1](https://github.com/Salil1129/Example29Aug24/blob/main/Scripts/template.sbatch.sh)


## Class proposal


[Proposal](https://github.com/Salil1129/BIOL7263/blob/main/class_presentation/Salil%20Jindal's%20Research%20proposal.pdf)

## Chapter 1 notes

 [Chapter 1](https://github.com/Salil1129/BIOL7263/blob/main/Chapter-1.docx)

## fastqc files

[ecoli_fastqc.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/fastqc/ecoli_fastqc.sbatch)

[ecoli_fastqc.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/fastqc/ecoli_fastqc.sh)

## Index files
[ecoli_index.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_index.sbatch)

[ecoli_index.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_index.sh)

## BWA files

[ecoli_bwa_mem.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_bwa_mem.sbatch)

[ecoli_bwa_mem.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_bwa_mem.sh)

## Converting SAM to BAM files
[ecoli_samtools_view.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_samtools_view.sbatch)

[ecoli_samtools_view.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_samtools_view.sh)

## Sorting BAM files

[ecoli_samtools_sort.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_samtools_sort.sbatch)

[ecoli_samtools_sort.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_samtools_sort.sh)

## Sorting BAM files by read name

[samtools_sort_name.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_sort_name.sbatch)

[samtools_sort_name.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_sort_name.sh)

## Adding tags to the file

[samtools_add_tags.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_add_tags.sbatch)

[samtools_add_tags.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_add_tags.sh)

## Re-sorting BAM file by chromosomal coordinates

[samtools_sort_chromo.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_sort_chromo.sh)

[samtools_sort_chromo.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_sort_chromo.sbatch)

## Removing duplicates

[samtools_duplication_removal.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_duplication_removal.sbatch)

[samtools_duplication_removal.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_duplication_removal.sh)

## Indexing the BAM file

[samtools_indexing.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_indexing.sbatch)

[samtools_indexing.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_indexing.sh)

## Mapping statistics (quali map)

[qualimap.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/qualimap.sh)

[qualimap.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/qualimap.sbatch)

##Automated Variant Calling

[ecoli_vf.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_vcf.sbatch)

[ecoli_vcf.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_vcf.sh)

## To find out variable sites

[ecoli_call.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_call.sbatch)

[ecoli_call.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_call.sh)

[ecoli_flit.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_filt.sbatch)

[ecoli_flit.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_filt.sh)

## Locating Genes that are Missing Compared to the Reference

[ecoli_cover.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_cover.sbatch)

[ecoli_cover.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_cover.sh)








