
#BIOL7263

This is the home for all of my work for my favorite class!

# Here are my notes from:

 [Day 1](https://github.com/Salil1129/Example29Aug24/blob/main/Scripts/template.sbatch.sh)


## Class proposal


[Proposal](https://github.com/Salil1129/BIOL7263/blob/main/class_presentation/Salil%20Jindal's%20Research%20proposal.pdf)

## Chapter 1 notes

 [Chapter 1](https://github.com/Salil1129/BIOL7263/blob/main/Chapter-1.docx)

## fastqc files

[ecoli_fastqc.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/fastqc/ecoli_fastqc.sbatch)

[ecoli_fastqc.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/fastqc/ecoli_fastqc.sh)

## Index files
[ecoli_index.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_index.sbatch)

[ecoli_index.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_index.sh)

## BWA files

[ecoli_bwa_mem.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_bwa_mem.sbatch)

[ecoli_bwa_mem.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_bwa_mem.sh)

## Converting SAM to BAM files
[ecoli_samtools_view.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_samtools_view.sbatch)

[ecoli_samtools_view.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_samtools_view.sh)

## Sorting BAM files

[ecoli_samtools_sort.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_samtools_sort.sbatch)

[ecoli_samtools_sort.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/ecoli_samtools_sort.sh)

## Sorting BAM files by read name

[samtools_sort_name.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_sort_name.sbatch)

[samtools_sort_name.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_sort_name.sh)

## Adding tags to the file

[samtools_add_tags.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_add_tags.sbatch)

[samtools_add_tags.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_add_tags.sh)

## Re-sorting BAM file by chromosomal coordinates

[samtools_sort_chromo.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_sort_chromo.sh)

[samtools_sort_chromo.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_sort_chromo.sbatch)

## Removing duplicates

[samtools_duplication_removal.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_duplication_removal.sbatch)

[samtools_duplication_removal.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_duplication_removal.sh)

## Indexing the BAM file

[samtools_indexing.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_indexing.sbatch)

[samtools_indexing.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/samtools_indexing.sh)

## Mapping statistics (quali map)

[qualimap.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/qualimap.sh)

[qualimap.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/BWA/qualimap.sbatch)

## Automated Variant Calling

[ecoli_vf.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_vcf.sbatch)

[ecoli_vcf.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_vcf.sh)

## To find out variable sites

[ecoli_call.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_call.sbatch)

[ecoli_call.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_call.sh)

[ecoli_flit.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_filt.sbatch)

[ecoli_flit.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_filt.sh)

## Locating Genes that are Missing Compared to the Reference

[ecoli_cover.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_cover.sbatch)

[ecoli_cover.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bcf/ecoli_cover.sh)

## To convert bam to fastq file

[bam_to_fasta.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bam_to_fasta.sh)

[bam_to_fasta](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/bam_to_fasta.sbatch)

## QC for the fastq files

[fastq_fastqc.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/unmapped/fastq_fastqc.sbatch)

[fastq_fastqc.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/unmapped/fastq_fastqc.sh)

## De-novo Assembly

[unmapped_spades.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/unmapped/unmapped_spades.sbatch)

[unmapped_spades.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/unmapped/unmapped_spades.sh)

## Assesment of Assemblies

[unmapped_quast.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/unmapped/unmapped_quast.sh)

[unmapped_quast.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/unmapped/unmapped_quast.sbatch)

## Obtain Open Reading Frames

[unmapped_orf.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/unmapped/unmapped_orf.sbatch)

[unmapped_orf.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/unmapped/unmapped_orf.sh)

## Alignment of unmapped contigs to the reference genome

[unmapped_blastn.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/unmapped/unmapped_blastn.sbatch)

[unmapped_blastn.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/unmapped/unmapped_blastn.sh)

## Run Open Reading Frames Through pfam_scan

[unmapped_pfam.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/unmapped/unmapped_pfam.sbatch)

[unmapped_pfam.sh](https://github.com/Salil1129/BIOL7263/blob/main/Scripts/unmapped/unmapped_pfam.sh)

## De novo Assembly Using Short Reads

[fullspades.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/full_spades.sh)

[fullspades.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/full_spades.sbatch)

## Checking the Assembly

[quast.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/quast.sh)

[quast.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/quast.sbatch)

## Index the contigs

[reads_index.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/reads_index.sh)

[reads_index/sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/reads_index.sbatch)

## align QC reads to contigs and output SAM file

[mapping.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/mapping.sh)

[mapping.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/mapping.sbatch)

## convert the SAM file to a BAM file

[sam_tobam.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/sam_tobam.sh)

[sam_tobam.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/sam_tobam.sbatch)

## sort the BAM file

[bam_sort.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/bam_sort.sh)

[bam_sort.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/bam_sort.sbatch)

## Index BAM

[index_bam.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/index_bam.sh)

[index_bam.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/index_bam.sbatch)

## Mapping statstistics

[qualimap.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/qualimap.sh)

[qualimap.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/qualimap.sbatch)

## BLAST search

[blast.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/qualimap.sbatch)

[blast.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/blast.sbatch)

## Annotation of de novo Assembled Contigs

[orfipy.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/orfipy.sh)

[orfipy.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/orfipy.sbatch)

[pfam.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/pfam.sh)

[pfam.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/pfam.sbatch)

[blastp.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/blast_p.sh)

[blastp.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter4/blast_p.sbatch)

## Hybrid de novo Assembly

## QC the data

[pseudo_fastqc.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter5/pseudo_fastqc.sh)

[pseudo_fastqc.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter5/pseudo_fastqc.sbatch)

## Trim the Reads

[pseudo_trim.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter5/pseudo_trim.sh)

[pseudo_trim.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter5/pseudo_trim.sbatch)

## Assembly

[pseudo_short_assembly](https://github.com/Salil1129/BIOL7263/blob/main/Chapter5/pseud_short_assembly.sh)

[pseudo_short_assembly.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter5/pseud_short_assembly.sbatch)

## Hybrid Assembly

[pseudo_long_assembly.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter5/pseudo_long_assembly.sh)

[pseudo_long_assembly.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter5/pseudo_long_assembly.sbatch)

## Align Reads Back to Reference

## Index the BAM file

[index_bam.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter5/index_bam.sh)

[index_bam.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter5/index_bam.sbatch)

## Mapping the Illumina reads

[illumina_map.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter5/illumina_map.sh)

[illumina_map.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter5/illumina_map.sbatch)

## Mapping the PacBio data to the assembly

[pacbio_map.sh](https://github.com/Salil1129/BIOL7263/blob/main/Chapter5/pacbio_map.sh)

[pacbio_map.sbatch](https://github.com/Salil1129/BIOL7263/blob/main/Chapter5/pacbio_map.sbatch)










